rust/src/main.rs

```diff
@@ -32,23 +61,33 @@ fn setup_logging() -> Result<(), fern::InitError> {
             _ => log::LevelFilter::Info,
         },
         // Default to info
-        Err(_) => log::LevelFilter::Info
+        Err(_) => log::LevelFilter::Info,
     };
 
-    fern::Dispatch::new()
+    let fern_dispatch = fern::Dispatch::new()
         .format(|out, message, record| {
             out.finish(format_args!(
                 "{}[{}][{}] {}",
                 chrono::Local::now().format("[%Y-%m-%d][%H:%M:%S]"),
                 record.target(),
                 record.level(),
-                message
+                message,
             ))
         })
         .level(level_filter)
-        .chain(std::io::stderr())
-        .chain(fern::log_file("xi-core.log")?)
-        .apply()?;
+        .chain(io::stderr());
+
+    if let Some(logging_file_path) = path_for_log_file(XI_LOG_FILE) {
+        fern_dispatch
+            .chain(fern::log_file(logging_file_path)?)
+            .apply()?;
+    } else {
+        eprintln!("[WARNING] There was an issue getting the path for the log file, using stdout");
+        fern_dispatch
+            .chain(io::stdout())
```

Comment:
we can't chain to `stdout`; it is being piped to the client, which will try and interpret it as RPCs. 😱

if `fern_dispatch` was declared `mut`, you could do this with just,
```rust
// {
// earlier stuff
    if let Some(logging_file_path) = path_for_log_file(XI_LOG_FILE) {
        fern_dispatch = fern_dispatch.chain(..);
    }
    info!("Logging with fern setup");
    fern_dispatch.apply()
}
```

or something similar.

Reply:
Aaaaaaahhhhhhh I see. Fixed 😅